libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -340,8 +340,9 @@ bool AP_Arming::gps_checks(bool report)
         if (first_unconfigured != AP_GPS::GPS_ALL_CONFIGURED) {
             if (report) {
                 GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_CRITICAL,
-                                                 "PreArm: GPS %d has not been fully configured",
-                                                  first_unconfigured);
+                                                 "PreArm: GPS %d failing configuration checks",
+                                                  first_unconfigured + 1);
```

Comment:
Good point on the %u. The bug was users perceptions. If you told a user that GPS 1 was failing configurations they would think that was the first GPS was the problem not the second one. The string was changed so that we could tell the user which GPS wasn't working (IE which firmware version is flashed).